helix-term/src/ui/menu.rs

```diff
@@ -84,13 +84,13 @@ impl<T: Item> Menu<T> {
                     let text = option.filter_text();
                     // TODO: using fuzzy_indices could give us the char idx for match highlighting
                     self.matcher
-                        .fuzzy_match(text, pattern)
-                        .map(|score| (index, score))
+                        .fuzzy_indices(text, pattern)
+                        .map(|(score, fuzzy_indices)| (index, score, fuzzy_indices))
```

Comment:
This is very slow and expensive for large datasets. We should do what we do in the file picker: use `fuzzy_match` to score, then use `fuzzy_indices` inside the render method to only compute the indices for elements on screen.

Reply:
In this case the `render` function relies on `Table::render_table` function. I can do that, but it would require passing more information to the latter (the highlighted style and the prompt), making it less general. Is it okay?

Reply:
Sorry for the delay, https://github.com/helix-editor/helix/pull/3053/files#diff-b6c51d6e6645b52604abc7b9cd90c0200ab42d52be2f7e623929376971e784c6R641-R715 has a similar implementation where it patches the spans while rendering. We should be able to reuse the code once that PR is merged

Reply:
I see, reusing the code would definitely be the best. Unfortunately that PR hasn't seen much progress in the last couple of months; if you prefer I can pick it up and continue where it was left.

Reply:
@danyspin97 #3053 has been merged! That should help this PR be able to move further.

Reply:
@LeoniePhiline Thanks for the heads up :) I have also updated #1905 so this PR should be close to being merged.